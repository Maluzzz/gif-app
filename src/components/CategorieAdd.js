import React, {useState} from "react"
import PropTypes from 'prop-types'

export const CategorieAdd = ({setCategories}) => {
  const [text, setText] = useState('')
  const handleChange = (e) => {
    setText(e.target.value)
    
  }
  const handleSubmit = (e) => {
      e.preventDefault()
      if(text.trim().length !== 0 ){
      setCategories(cat => [text, ...cat])
      setText('')
    }
  }
  return (
   
      <form onSubmit={handleSubmit} >
        <input type="text" value={text} onChange={handleChange} placeholder="Search for Gifs"/>
        <button >Search</button>
      </form>
  
  )
}

CategorieAdd.protoTypes = {
    setCategories: PropTypes.func.isRequired
}