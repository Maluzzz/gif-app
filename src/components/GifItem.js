import React from 'react'
import PropTypes from 'prop-types'

export default function GifItem({ title, url }) {
    return (
        <>
            <div className="card-image animate__animated  animate__pulse">
                <img src={url} alt={title}></img>
                <p className='gif-title'>{title}</p>
            </div>
        </>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}
