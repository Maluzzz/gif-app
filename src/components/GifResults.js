import React  from 'react'
import PropTypes from 'prop-types'

import { useFetchGifs } from '../hooks/useFetchGifs'
import GifItem from './GifItem'



export const GifResults = ({category}) => {

  const {data: imgs,loading} = useFetchGifs(category)


  return (
    <>
      <h3>{category}</h3>
      {loading && <p>'Cargando'</p>}
      <div className='card-container'>
        {imgs.map((img) => (
          <GifItem key={img.id} {...img} />
        ))}
      </div>
    </>
  )
}

GifResults.propTypes= {
  category: PropTypes.string.isRequired
}
