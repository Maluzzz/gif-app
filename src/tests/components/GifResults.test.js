import '@testing-library/jest-dom'

import { shallow } from 'enzyme'
import { useFetchGifs } from '../../hooks/useFetchGifs'
import { GifResults } from '../../components/GifResults'
jest.mock('../../hooks/useFetchGifs')

describe('Testing Gif Results Component', () => {

      const category = ''
      

      test('should render the first time without category', () => {
            useFetchGifs.mockReturnValue({
                  data: [],
                  loading: true
            })
            const wrapper = shallow(<GifResults category={category} />)
            expect(wrapper).toMatchSnapshot()
      })

      test('should render with information from useFetchGifs', () => {
            const gifs = [{
                  id: 'abcd',
                  title: 'Hello',
                  url: 'image.jpg'
            }]

            useFetchGifs.mockReturnValue({
                  data: gifs,
                  loading: false
            })
            const wrapper = shallow(<GifResults category={category} />)
            expect(wrapper).toMatchSnapshot()
            expect(wrapper.find('p').exists()).toBe(false)
            expect(wrapper.find('GifItem').length).toBe(gifs.length)
       
      })
    

})
