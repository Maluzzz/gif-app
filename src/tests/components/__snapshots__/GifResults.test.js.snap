// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Gif Results Component should render the first time without category 1`] = `
<Fragment>
  <h3 />
  <p>
    'Cargando'
  </p>
  <div
    className="card-container"
  />
</Fragment>
`;

exports[`Testing Gif Results Component should render with information from useFetchGifs 1`] = `
<Fragment>
  <h3 />
  <div
    className="card-container"
  >
    <GifItem
      id="abcd"
      key="abcd"
      title="Hello"
      url="image.jpg"
    />
  </div>
</Fragment>
`;
